### DISCUSSION TITLE
Local development environment not letting me into a room

### DISCUSSION THREAD

**Discussion by johnperezdev** [mozilla/hubs#4286] [Hubs Cloud Questions]
I am currently trying to set up a development environment for hubs, and all of the steps work fine up until I actually try to join a room on a local instance. Whenever I run npm start and try to create a new room or join an existing one that I made on the cloud site, I am met with 

![image](https://user-images.githubusercontent.com/55162758/119368490-1390ec80-bc81-11eb-8cd8-c9dc78aa7e7c.png)

and then a few moments later it says this: 

![image](https://user-images.githubusercontent.com/55162758/119368453-083dc100-bc81-11eb-8118-18b0979f9957.png)

Also it's worth mentioning that whenever I try to join a room that I created on the cloud, it says that I left and joined. So clearly it's at least trying to connect. 

![image](https://user-images.githubusercontent.com/55162758/119369037-b77a9800-bc81-11eb-8ad4-e566cc9faabd.png)

So does anyone know what I'm doing wrong? I know that this is kind of vague, but everything was done fresh. The cloud instance was set up exactly as instructed and so was the local portion. I'm at a complete loss at this point so any guidance is welcome. 
Thanks in advance. 

(Also does anyone know how to access the development portion of the hubs discord? I can't find it listed anywhere)

Edit:

Occasionally the command `npm ci` does not download all of the essential files. This causes the hubs test environment to not work. 
The solution I've seen the most success with is running the command again and praying.

**Comment by lihong-xmuseum** [comment]
I have the same problem. Have you found a solution?

**Reply by johnperezdev** [reply]
Kind of. My environment started working, but I'm genuinely not sure why (which is why I didn't post a solution). 

Some things that I think caused mine to start working are that I had a public room already made on my clouds site and I am just using a link directly to that room (https://localhost:8080/hub.html?hub_id=ExistingHubID instead of the link that the console tells you to go to). 

Also try different browsers. I know that my local environment only works on Firefox. It might be the same for you as well. 

Lastly, stick this line `process.env.HOST_IP = "localhost";` at the top of the file "../hubs/webpack.config.js". Even if this doesn't fix your current issue, it will solve a different issue of entities not loading when you're trying to test things later (once you have this issue fixed.) 

After that the only thing left I can recommend is to try to allow the CORS request through for localhost:8080 on your AWS instance. That is what I originally thought the issue was, but then mine just started working without me changing it so who knows.

**Reply by lihong-xmuseum** [reply]
Thank you very much. I'll try

**Comment by shamsail** [comment]
Don't run 
```
npm start
```
and 
```
npm run login
```
if you want to enter the room in local development then run
```
npm run dev
```
in case if you are logged in before, try to `logout` before running this command `npm run dev` by this
```
npm run logout
```
Also, one more thing, in case, if to enter the room is not working after running this command `npm run dev` without login so try to clear the cookies and cache.. as `npm run dev` does not work with `npm run login` and the cookies/cache related to this.

Here is one more thing,
`npm start` with `npm run login` will give you the rooms that are on your live server (but can't enter from local)
`npm run dev` without login will not give you the room that are on live server (but it will give you the access to enter the room with default room) (this is used when you want to do some code and test inside the scene/room layouts)

hope it will resolve your issues

**Reply by lihong-xmuseum** [reply]
No, try it, my local environment still can't create a room.

**Reply by johnperezdev** [reply]
Do you mind posting a screenshot of the errors the browser console shows?

**Reply by lihong-xmuseum** [reply]
![1626067822(1)](https://user-images.githubusercontent.com/69456381/125235716-7a439580-e315-11eb-9c25-ada05979597b.png)
![1626067851(1)](https://user-images.githubusercontent.com/69456381/125235742-816aa380-e315-11eb-8779-2704fac7bf59.png)

**Reply by johnperezdev** [reply]
Does your node_modules folder contain a folder called "naf-janus-adapter"? I realized that npm ci will not download it for some people.

**Reply by lihong-xmuseum** [reply]
Yes, there is this folder

**Reply by lihong-xmuseum** [reply]
I was able to run the local environment after re-executing 'npm ci', but it stopped working a day later. It's weird.
